src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -483,7 +483,8 @@ size_t runtime::gpu::CUDAEmitter::build_avg_pool(const GPURuntimeContext* ctx,
                     if (include_pad == false)
                     {
                         // count the number of (non-padded) elements
-                        writer << "pool_size += __popc(__ballot(within_tensor_bounds));\n";
+                        writer << "pool_size += __popc(__ballot_sync(0xffffffff, "
```

Comment:
Note, in the future, we can check if we can use mask to mask out some of the shift, memory load/store is not really full parallel. might save some work. No need to investigate for now.